lib/TwaManifest.js

```diff
@@ -0,0 +1,197 @@
+   * values for those fields are valid.
+   */
+  validate() {
+    if (!this.host) {
+      return false;
+    }
+
+    if (!this.name) {
+      return false;
+    }
+
+    if (!this.startUrl) {
+      return false;
+    }
+
+    if (!this.iconUrl) {
+      return false;
+    }
+
+    return true;
+  }
+
+  _hexColor(color) {
+    const rgbColor = colorString.get.rgb(color);
+    return colorString.to.hex(rgbColor);
+  }
+
+  themeColorHex() {
+    return this._hexColor(this.themeColor);
+  }
+
+  navigationColorHex() {
+    return this._hexColor(this.navigationColor);
+  }
+
+  backgroundColorHex() {
+    return this._hexColor(this.backgroundColor);
+  }
+
+  /**
+   * Creates a new TwaManifest, using the URL for the Manifest as a base URL and uses the content
+   * of the Web Manifest to generate the fields for the TWA Manifest.
+   *
+   * @param {URL} webManifestUrl the URL where the webmanifest is available
+   * @param {WebManifest} the Web Manifest, used as a base for the TWA Manifest.
+   * @returns {TwaManifest}
+   */
+  static fromWebManifestJson(webManifestUrl, webManifest) {
+    const icon = util.findSuitableIcon(webManifest, 'any');
+    const maskableIcon = util.findSuitableIcon(webManifest, 'maskable');
+    const fullStartUrl = new URL(webManifest['start_url'] || '/', webManifestUrl);
+
+    const twaManifest = new TwaManifest({
+      packageId: generatePackageId(webManifestUrl.host),
+      host: webManifestUrl.host,
+      name: webManifest['short_name'] || webManifest['name'] || 'My TWA',
+      themeColor: webManifest['theme_color'] || '#FFFFFF',
+      navigationColor: webManifest['theme_color'] || '#FFFFFF',
+      backgroundColor: webManifest['background_color'] || '#FFFFFF',
+      startUrl: fullStartUrl.pathname + fullStartUrl.search,
+      iconUrl: icon ? new URL(icon.src, webManifestUrl).toString() : undefined,
+      maskableIconUrl:
+         maskableIcon ? new URL(maskableIcon.src, webManifestUrl).toString() : undefined,
+      appVersion: '1.0.0',
+      signingKey: {
+        path: './android.keystore',
+        alias: 'android',
+      },
+      splashScreenFadeOutDuration: 300,
+      useBrowserOnChromeOS: true,
+      enableNotifications: false,
+    });
+    return twaManifest;
+  }
+
+  /**
+   * Fetches a Web Manifest from the url and uses it as a base for the TWA Manifest.
+   *
+   * @param {String} url the URL where the webmanifest is available
+   * @returns {TwaManifest}
+   */
+  static async fromWebManifest(url) {
+    const webManifest = await fetch(url).then((res) => res.json());
+    const webManifestUrl = new URL(url);
+    return TwaManifest.fromWebManifestJson(webManifestUrl, webManifest);
+  }
+
+  /**
+   * Loads a TWA Manifest from the file system.
+   *
+   * @param {String} fileName the location of the TWA Manifest file
+   */
+  static async fromFile(fileName) {
+    const json = JSON.parse(await fs.promises.readFile(fileName));
```

Comment:
Just to check my understanding - if readFile throws an error (since the file doesn't exist), that error will be thrown from here?

Reply:
Also, do we want some slightly nicer error handling if the file has invalid JSON?

Reply:
If `readFile` fails, an error will be throw. The same thing happens if `JSON.parse` receives and invalid JSON.  Friendly errors are also something we need to look from a more general perspective. Filed #32 to track this.